metadoc-js/src/main/scala/metadoc/ScalaDocumentSymbolProvider.scala

```diff
@@ -0,0 +1,69 @@
+package metadoc
+
+import scala.{meta => m}
+import scala.meta.inputs.Input
+import scala.meta.Attributes
+import scala.meta.Denotation
+import scala.scalajs.js
+import scala.scalajs.js.annotation._
+import metadoc.schema.Index
+import metadoc.schema.Position
+import monaco.CancellationToken
+import monaco.editor.IReadOnlyModel
+import monaco.languages.DocumentSymbolProvider
+import monaco.languages.SymbolInformation
+import monaco.languages.SymbolKind
+import org.scalameta.logger
+
+@ScalaJSDefined
+class ScalaDocumentSymbolProvider(attrs: Attributes, index: Index)
+    extends DocumentSymbolProvider {
+  override def provideDocumentSymbols(
+      model: IReadOnlyModel,
+      token: CancellationToken
+  ) = {
+    val denotations = attrs.denotations.map { case (s, d) => s -> d }.toMap
+    val symbols = for {
+      sym <- index.symbols
+      denotation <- denotations.get(m.Symbol(sym.symbol))
+      if {
+        import denotation._
+        (!isPARAM && !isTypeParam) && {
+          isClass ||
+          isTrait ||
+          isObject ||
+          isDef ||
+          isVal
+        }
+      }
+      kind <- symbolKind(denotation)
```

Comment:
We are filtering twice here, since anything we cannot map to a SymbolKind won't be shown. I suggest we change the above filter to only remove parameters and type parameters.